x/delayedack/keeper/invariants.go

```diff
@@ -0,0 +1,115 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	commontypes "github.com/dymensionxyz/dymension/v3/x/common/types"
+	"github.com/dymensionxyz/dymension/v3/x/delayedack/types"
+)
+
+// RegisterInvariants registers the delayedack module invariants
+func RegisterInvariants(ir sdk.InvariantRegistry, k Keeper) {
+	ir.RegisterRoute(types.ModuleName, "rollapp-finalized-packet", PacketsFromFinalizedHeightsAreFinalized(k))
+	ir.RegisterRoute(types.ModuleName, "rollapp-reverted-packet", PacketsFromRevertedHeightsAreReverted(k))
+}
+
+// AllInvariants runs all invariants of the x/delayedack module.
+func AllInvariants(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		res, stop := PacketsFromFinalizedHeightsAreFinalized(k)(ctx)
+		if stop {
+			return res, stop
+		}
+		res, stop = PacketsFromRevertedHeightsAreReverted(k)(ctx)
+		if stop {
+			return res, stop
+		}
+		return "", false
+	}
+}
+
+// PacketsFromFinalizedHeightsAreFinalized checks that all rollapp packets stored for a rollapp finalized height are also finalized
```

Comment:
As I see it the invariants try to cover 2 cases:
1. Rollapp packet (RP) is finalized but Rollapp height (RH) is not finalized
```
if (latestFinalizedHeight == 0 || packet.ProofHeight > latestFinalizedHeight) && packet.Status == commontypes.Status_FINALIZED {
				msg += fmt.Sprintf("rollapp packet for the height should not be in finalized status. height=%d, rollapp=%s, status=%s\n", packet.ProofHeight, packet.RollappId, packet.Status)
				return msg, true
			}
```
2. RP is not finalized but RH is finalized
```
if packet.ProofHeight <= latestFinalizedHeight && packet.Status != commontypes.Status_FINALIZED {
				msg += fmt.Sprintf("rollapp packet for the height should be in finalized status. height=%d, rollapp=%s, status=%s\n", packet.ProofHeight, packet.RollappId, packet.Status)
				return msg, true
			}
```

my 2 cents here is that:
1. the invariant name is confusing so either we make it include both cases or we split it to 2 different invariants
2. not sure why the comparison to 0 is necessary when iterating vs just checking those 2 cases, i.e 

```
if packet.ProofHeight > latestFinalizedHeight && packet.Status == commontypes.Status_FINALIZED {
				msg += fmt.Sprintf("rollapp packet for the height should not be in finalized status. height=%d, rollapp=%s, status=%s\n", packet.ProofHeight, packet.RollappId, packet.Status)
				return msg, true
			}
			if packet.ProofHeight <= latestFinalizedHeight && packet.Status != commontypes.Status_FINALIZED {
				msg += fmt.Sprintf("rollapp packet for the height should be in finalized status. height=%d, rollapp=%s, status=%s\n", packet.ProofHeight, packet.RollappId, packet.Status)
				return msg, true
			}
```

Reply:
improved it according to the comments